documentation/internal/Filters-for-Applications.md

```diff
@@ -0,0 +1,292 @@
+Considered in implementation: **Yes** [#131 - Headers]
+Can have combined values: **Yes** `sort-by=stage:asc,application-name:desc`
+
+Options:
+
+- `column-name:direction`
+- direction [Optional] `asc`/`desc`
+- if no direction is included use default: `asc`
+
+##### user-role
+
+**This is a **compulsory** filter.**
+In case none is included in URL will be deduced from permission of type in user's persmissions.
+
+Example: `user-role=applicant`
+Considered in implementation: **Yes** [#131 - Headers]
+Can have combined values: **No**
+
+Options:
+
+- Deduced by user's permissions (**TODO: More detailed explanation of user-role per permissions**)
+
+##### outcome
+
+Example: `outcome=pending`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `outcome=pending,approved`
+
+Options:
+
+- Set of `outcome` common for all templates (application types)
+
+##### action
+
+Example: `action=edit-draft`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `action=edit-draft,`
+
+Options:
+
+- Applicant:
+  `edit-draft`
+  `make-updates`
+  `renew`
+  `view` (Submitted)
+- Reviewer
+  `start-review`
+  `continue-review`
+  `review-updates`
+  `view` (Submitted)
+- Consolidator
+  `assign`
+  `consolidate`
+  `review-updates`
+  `view` (Submitted)
+
+---
+
+#### String filters:
+
+##### assigned
+
+Example: `assigned=none`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `assigned=none,"Carl"`
+
+Options:
+
+- `none` - Applications with sections not-assigned
+- `"username"` - Reviewer username
+
+##### consolidator
+
+Example: `consolidator="Tony"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `consolidator="Tony","Andrei"`
+
+Options:
+
+- `"username"` - Consolidator username
+
+##### applicant
+
+Example: `applicant="John"`
+Considered in implementation: **No**
+Can have combined values: **Yes** e.g: `applicant="John","Nicole"`
+
+Options:
+
+- `"username"` - Applicant username
+
+##### org
+
+Example: `org="Company A"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **Yes** e.g: `org="Company A","Company B"`
+
+Options:
+
+- `"organisation name"`
+
+##### search
+
+Example: `search="abc 123"`
+Considered in implementation: **Yes** [#251 - Filters]
+Can have combined values: **No**
+
+Options:
+
+- String containing [A-Z], [a-z], [0-9], _space_
+- Uses _starts with_ on search including following columns: "Application name", "Stage", ... (**TODO: continue listing columns**)
+- (See more rules on Formats & Rules)
+
+---
+
+#### Date filters:
+
+##### last-active-date
+
+Example: `last-active-date=2021-01-01`
+Considered in implementation: **Yes** [#228 - Dates in columns]
+Can have combined values: **Yes** e.g: `last-active-date=today,last-week`
+
+Options:
+
+- Pre-defined string: `today` (See more formats on Formats & Rules)
```

Comment:
I assume (for implementation) we'd just have a look-up table with a set of functions to convert pre-defined strings into the standard `YYYY-MM-DD:YYYY-MM-DD` format for further processing?

Reply:
I was thinking about just storing that on the Front-end. And converting on the front-end before calling the query.